src/engine.ml

```diff
@@ -537,14 +537,37 @@ let kex_server config auth_user_pass session (my_key_material : my_key_material)
     tls data =
   let open Result.Syntax in
   let* their_tls_data = Packet.decode_tls_data ~with_premaster:true data in
-  let* () =
+  let* authenticated =
     match (auth_user_pass, their_tls_data.user_pass) with
-    | None, _ -> Ok ()
-    | Some _, None -> Error (`Msg "no username and password provided")
+    | None, _ -> Ok true (* erm... *)
```

Comment:
this case needs to be there as it is, since there could be servers that only use `verify-client-cert` -- so the server must be fine with no auth-user-pass being provided.

we can of course when creating the server check and warn if verify-client-cert is 'none' and auth-user-pass is not provided (and print a big warning that no authentication will take place).

Reply:
done in 238c2e9dea4ec7d20096d591e1865da261ced55d